README.md

```diff
@@ -1,277 +1,1037 @@
+batcher based composition to *not be trustless* (without the composition logic, more DApps can fit
+in a transaction).
+- No delegation control or voting control while using the DEX - DEXs try to issue governance tokens
+to get around this but fair distribution of the governance tokens is rarely accomplished (it is a
+very hard problem). Even if the governance tokens were distributed fairly, they are largely just for
+show since the DEX creators can still choose to go against the vote. There is no trustless
+connection between the governance tokens and what actually happens with the stake/DApp. Finally,
+even if there was a trustless connection, there is no way to actually have all users express their
+own stake preferences when the stake is decided democratically. It is almost inevitable that the
+voting outcome will go against the wishes of some users. *This is in direct violation of the
+security assumptions of PoS.* No matter how you look at it, LP based DApps are an existenstial
+problem for PoS blockchains. As LP based DApps become more adopted, the underlying PoS blockchain
+will only become more centralized. LPs can never be used as the foundation for a trustless and
+decentralized PoS blockchain economy.
+
+DEXs that hope to become the foundation for a healthy and fully trustless, decentralized economy
+must adhere to a radically different approach that takes full advantage of the composability and
+availability guarantees offered by the eUTxO model while not undermining the security of the very
+blockchain the economy is built on.
+
+## The Cardano-Swaps Protocol
+
+Cardano-Swaps is an order-book based DEX that takes inspiration from Axo's programmable swaps
+[design](https://www.axo.trade/whitepaper.pdf), adds delegation control as a foundational feature,
+naturally incentivizes its own liquidity, and, through the use of Beacon Tokens, removes the need for
+specialized indexers. In short, it solves all of the pressing issues that made order-book DEXs
+inferior to LP based DEXs. 
+
+### High-Level Overview
+
+The protocol is comprised of two types of composable swaps: a one-way swap and a two-way swap.
+One-way swaps are effectively just limit orders (ie, they can only be executed at the specified
+price or better). Most users will likely use the one-way swap.
+
+Two-way swaps are a mechanism for profitably providing liquidity to the protocol without having to
+resort to yield farming. These two-way swaps can go in either direction (eg, a two-way ADA <-> ERGO
+swap can go ADA -> ERGO or ERGO -> ADA). By allowing the liquidity provider to specify separate
+prices for each direction, they can profit from either direction of the swap. For example, if the
+market rate for DJED <-> USDC is 1:1, Alice can set DJED -> USDC to 1.01 USDC per DJED and USDC ->
+DJED to 1.01 DJED per USDC. This means Alice makes a 1% return no matter what direction is used.
```

Comment:
@rphair should these be:

> Alice can set DJED -> USDC to 1.01 usdc per djed...

Reply:
They're being quoted as trading symbols & in exchange rates, so the convention would be to uppercase them.